src/main/java/com/project/gamestore/controllers/RAWGController.java

```diff
@@ -62,20 +62,41 @@ public void processTitle(String title) {
 
     @GetMapping("/videogames/search")
     public List<VideoGame> ListGamesByName(@RequestParam String title) throws JsonProcessingException {
-        // Process title
+        // Process title.
         processTitle(title);
 
-        // Call the RAWG API to get a certain number of games
+        String url = "https://api.rawg.io/api/games?search=" + title + "&page_size=10&key=79fc5d7fcd144b99ade6f0aafc6e8c74";
+
+        // Call the RAWG API to get a certain number of games.
         ResponseEntity<String> response = restTemplate.getForEntity(
-                "https://api.rawg.io/api/games?search=" + title + "&page_size=10&key=79fc5d7fcd144b99ade6f0aafc6e8c74",
+                url,
                 String.class);
         String jsonString = response.getBody();
 
-        // Convert the json string to a json object, so we can access data.
-        JSONObject json = new JSONObject(jsonString);
-        JSONArray gamesList = json.getJSONArray("results");
-  
-        List<VideoGame> games = new ObjectMapper().readValue(gamesList.toString(), new TypeReference<List<VideoGame>>() {});
+        // Convert the json string to a json node, so we can access data.
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode rootNode = objectMapper.readValue(jsonString, JsonNode.class);
+
+        // Move into the array with the list of games.
+        JsonNode gamesList = rootNode.get("results");
+
+        // Iterate through the games and retrieve the game info.
+        List<VideoGame> games = new ArrayList<>();
+        for (int i = 0; i < gamesList.size(); i++) {
+            VideoGame g = new VideoGame();
+            g.setId(Integer.parseInt(String.valueOf(gamesList.get(i).get("id"))));
```

Comment:
Remove the parsing and valueOf. The string data is saved as JSON string. We don't want that. Instead use JsonNode methods used to make the conversion.

- .asText()
- .asInt()
- .asDouble()